civis/base.py

```diff
@@ -84,3 +85,81 @@ def _call_api(self, method, path=None, params=None, data=None, **kwargs):
             resp = convert_response_data_type(resp,
                                               return_type=self._return_type)
             return resp
+
+
+FINISHED = ['success', 'succeeded']
+FAILED = ['failed']
+NOT_FINISHED = ['queued', 'running']
+CANCELLED = ['cancelled']
+DONE = FINISHED + FAILED + CANCELLED
+
+# Translate Civis state strings into `future` state strings
+STATE_TRANS = {}
+for name in FINISHED + FAILED:
+    STATE_TRANS[name] = futures._base.FINISHED
+for name in NOT_FINISHED:
+    STATE_TRANS[name] = futures._base.RUNNING
+for name in CANCELLED:
+    STATE_TRANS[name] = futures._base.CANCELLED_AND_NOTIFIED
+
+
+class CivisAsyncResultBase(futures.Future):
```

Comment:
Would it be clearer to move some of the comments directly below the docstring for the `PollableResult` class here? Also worth mentioning that subclasses should override `_check_result` or `set_result`/`set_exception`. Are those the only methods that need to be implemented?